x-pack/plugins/lens/public/indexpattern_datasource/indexpattern.tsx

```diff
@@ -574,6 +575,16 @@ export function getIndexPatternDatasource({
             timeRange
           ),
         getVisualDefaults: () => getVisualDefaultsForLayer(layer),
+        getMaxPossibleNumValues: (columnId) => {
+          // TODO - is the isReferenced check necessary?
+          if (layer && layer.columns[columnId] && !isReferenced(layer, columnId)) {
```

Comment:
just curious - why isReferenced check?

Reply:
Copied from `getOperationForColumnId`. I'm not sure if it makes sense here

Reply:
cc @flash1293

Reply:
No, I'd say it can be removed

Reply:
This is used in some places because there are intermediary columns (like when using formula) which are not exposed to the visualization. But it's not relevant for this logic, so let's leave it out so it won't confuse anyone